subsys/net/ip/net_if.c

```diff
@@ -1623,6 +1623,100 @@ bool net_if_ipv4_addr_rm(struct net_if *iface, struct in_addr *addr)
 
 	return false;
 }
+
+static struct net_if_mcast_addr *ipv4_maddr_find(struct net_if *iface,
+						 bool need_free,
+						 const struct in_addr *addr)
+{
+	int i;
+
+	for (i = 0; i < NET_IF_MAX_IPV4_MADDR; i++) {
+		if (need_free && iface->ipv4.mcast[i].is_used) {
```

Comment:
I would have made s/need_free/is_used and made one test only here, as it was semantically better because maddr_rm or maddr_lookup for instance are looking for used ones, it's not like they "need" a non-free.
The "need" keyword here bugs me a bit. Fits maddr_add, but is_used was also proper here.

But ok, as you want.

Reply:
Ok, I will change the naming here.